src/OData.QueryBuilder/Expressions/Visitors/ODataOptionFilterExpressionVisitor.cs

```diff
@@ -78,7 +78,16 @@ protected override string VisitMethodCallExpression(MethodCallExpression methodC
                     return $"{all0}/{nameof(IODataOperator.All).ToLowerInvariant()}({all1})";
                 case nameof(IODataOperator.Any):
                     var any0 = VisitExpression(methodCallExpression.Arguments[0]);
-                    var any1 = VisitExpression(methodCallExpression.Arguments[1]);
+                    var any1 = "";
+
+                    if (methodCallExpression.Arguments.Count > 1)
```

Comment:
I tried to do `if (methodCallExpression.Arguments.Count > 1 && methodCallExpression.Arguments[1] != null)`, but couldn't figure out how to access the value of the argument. So in the end I just cleared `any1` if it was `null`. There might be a better way to do this though.

Reply:
at the moment, this is the only way, but it should be
```csharp
var any0 = VisitExpression(methodCallExpression.Arguments[0]);
var any1 = default(string);

if (methodCallExpression.Arguments.Count > 1)
{
    any1 = VisitExpression(methodCallExpression.Arguments[1]);

    if (any1.IsNullOrQuotes())
    {
        if (!_odataQueryBuilderOptions.SuppressExceptionOfNullOrEmptyOperatorArgs)
        {
            throw new ArgumentException("Func is null");
        }
    }
}

return $"{any0}/{nameof(IODataOperator.Any).ToLowerInvariant()}({any1})";
```

earlier, I forgot to check for null for the Func <T, bool> func parameter.

tests need to check for null
https://github.com/ZEXSM/OData.QueryBuilder/blob/main/test/OData.QueryBuilder.Test/ODataQueryCollectionTest.cs#L741
https://github.com/ZEXSM/OData.QueryBuilder/blob/main/test/OData.QueryBuilder.Test/ODataQueryCollectionTest.cs#L823

Reply:
Right, I'll update it, thanks! Should any1 still be set to `default(string)` if `SuppressExceptionOfNullOrEmptyOperatorArgs` is `true`? As generating a query like `Collection/any(null)` is invalid.

Reply:
I ended up doing the same as the In operator, returning default if errors are suppressed, to avoid generating invalid query.